ecmd-core/ext/fapi2/capi/target.H

```diff
@@ -303,6 +304,49 @@ Target<K, M, V>::getChildren(const TargetFilter i_filter,
     return l_children;
 }
 
+template<TargetType K, MulticastType M, typename V>
+template< TargetType T >
+inline ReturnCodes
+Target<K, M, V>::reduceType(Target<T, M, V>& o_target) const
+{
+    check_reduceType<T>();
+
+    bool compatible = true;
+    plat_target_handle_t l_handle = this->get();
+    const ecmdChipTarget * l_pEcmdTarget = l_handle;
+    fapi2::TargetType l_type = TARGET_TYPE_NONE;
+
+    uint32_t l_rc = fapi2GetTargetType(*l_pEcmdTarget, l_type);
+    if (l_rc)
+    {
+        compatible = false;
+    }
+
+    // Here you would check that the platform target's type is compatible with T,
+    // also taking multicast into account (iff the plat target is a real multicast
```

Comment:
typo on "if"

Reply:
It's shorthand for "if and only if": https://en.wikipedia.org/wiki/If_and_only_if

Reply:
But now that I think about it, incorrectly so since T may include MULTICAST if the plat target is not a real multicast target. You're right, should be "if".